src/ucs/config/parser.c

```diff
@@ -1423,47 +1423,38 @@ static ucs_status_t ucs_config_parser_get_sub_prefix(const char *env_prefix,
 
 void ucs_config_parse_config_files()
 {
-    char config_files[5][MAXPATHLEN] = {{0}};
-    char *dlpath_dup = NULL, *ucx_config_dir = NULL;
-    int i;
-    Dl_info ucs_dl_info;
+    char path[MAXPATHLEN] = {0};
+    const char *tmp = NULL;
 
     /* Global config dir */
-    strcpy(config_files[0], UCX_GLOBAL_CONFIG_FILE);
+    ucs_config_parse_config_file(UCX_GLOBAL_CONFIG_FILE, 1);
 
     /* Library dir */
-    if (dladdr(ucs_config_parse_config_files, &ucs_dl_info) != 0) {
-        dlpath_dup = ucs_strdup(ucs_dl_info.dli_fname, "config_parser");
-        snprintf(config_files[1], MAXPATHLEN, "%s/../etc/%s",
-                 dirname(dlpath_dup), UCX_CONFIG_FILE_NAME);
+    tmp = ucs_sys_get_lib_path();
+    if (tmp != NULL) {
+        ucs_snprintf_safe (path, MAXPATHLEN, "%s/../etc/%s",
+                 tmp, UCX_CONFIG_FILE_NAME);
+        ucs_config_parse_config_file(path, 1);
     }
 
     /* User home dir */
-    snprintf(config_files[2], MAXPATHLEN, "%s/%s", getenv("HOME"),
-             UCX_CONFIG_FILE_NAME);
-
-    /* Custom conf dir */
-    ucx_config_dir = getenv("UCX_CONFIG_DIR");
-    if (ucx_config_dir != NULL) {
-        snprintf(config_files[3], MAXPATHLEN, "%s/%s", ucx_config_dir,
+    tmp = getenv("HOME");
```

Comment:
tmp -> env_value / env_str /..

Reply:
Renamed, chose another name. `env` is not fully correct, as "library dir" doesn't come from env.